Assets/Scripts/Augment/BulletController.cs

```diff
@@ -14,34 +16,62 @@ public class BulletController : ProjectileController
     private float bulletSpeed = 50f;
 
     private VFXTextureFormatter trailPosTexture;
-    //private Texture2D trailPosTexture;
-    //private float[] trailPositions;
 
     [SerializeField]
     private VisualEffect trail;
 
     [SerializeField]
     private VisualEffect flash;
 
+    [SerializeField]
+    private BarrelAnimator animator;
+
     private ProjectileState projectile = new ProjectileState();
 
 
     protected override void Awake()
     {
         base.Awake();
         UpdateProjectileMovement += ProjectileMotions.MoveWithGravity;
-        //trailPositions = new float[vfxPositionsPerSample * collisionSamples * 3];
         trailPosTexture = new VFXTextureFormatter(vfxPositionsPerSample * collisionSamples);
         trail.SetTexture("Position", this.trailPosTexture.Texture);
+
+        // TODO should be in projectilecontroller :)
+        gunController = transform.parent.GetComponent<GunController>();
+        if (!gunController)
+        {
+            Debug.Log("HatBarrel not attached to gun parent!");
+            return;
+        }
+        gunController.onInitializeGun += OnInitialize;
+        //gunController.onFire += OnFire;
```

Comment:
Why is onFire removed? 🤔

Reply:
It was replaced with a call to InitializeProjectile (and not invoked anymore), at this point we could actually subscribe to that delegate again though >_>
and I may have just copied this over from HatBarrel without putting too much thought into it
and these names give me pain btw :))))

Reply:
There, it's cleaned up now... Though please test the game, seems like I broke the hat barrel on my end :/

Reply:
ok thanks for fixing my mistake fredrik